packages/react/src/components/UIShell/SideNav.js

```diff
@@ -107,11 +108,19 @@ const SideNav = React.forwardRef(function SideNav(props, ref) {
         handleToggle(event, true);
       }
     };
+
     eventHandlers.onBlur = (event) => {
       if (!event.currentTarget.contains(event.relatedTarget)) {
         handleToggle(event, false);
       }
     };
+
+    eventHandlers.onKeyDown = (event) => {
+      if (match(event, keys.Escape)) {
+        handleToggle(event, false);
+        window.location.href = '#main-content';
```

Comment:
I'm not sure what the right thing to do here is 🤔, like is it worth adding an `href` prop like [skipToContent](https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/UIShell/SkipToContent.js) or should we just drop it and not redirect to the main content at all?

Ref:   https://github.com/carbon-design-system/carbon/issues/13420
> This should be an easy fix (just need to decide where focus would go, which I think is the main content, probably at top of region)

Reply:
Yeah that was my thought - match the setup of SkipToContent and allow it to be configurable. I think we should provide something out of the box for it as a reasonable default

Reply:
I added an `href` prop just like `SkipToContent.js`. 

The `SkipToContent.js` functionality appears to be functioning properly, but I've noticed a weird behaivour when using MacOS's voice over feature. Specifically, pressing the ESC key to skip to the main content doesn't trigger the voice over to inform the user that they've reached the main content, even though they are at the main content.

Both on `SkipToContent.js` and `SideNav`